enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,342 @@
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce new Insights `CustomResourceDefinition` types in the OpenShift API providing a way to configure the Insights operator and to report its status. 
+
+The proposal for the new `insights.config.openshift.io` resource is:
+
+```go
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	// GatheringConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig *GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+}
+
+type GatheringConfig struct {
+	// DataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	// +kubebuilder:default=NoPolicy
+	DataPolicy DataPolicy `json:"dataPolicy"`
+	// ForceGatherReason enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	// When the forced gathering is finished then the value is cleared.
+	ForceGatherReason string `json:"forceGatherReason"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "NoPolicy"
+	// IP addresses and cluster domain name is obfuscated
+	IPsAndClusterDomainPolicy DataPolicy = "IPsAndClusterDomainPolicy"
+)
+
+type DataPolicy string
+```
+Note that the `support` secret configuration still takes precedence over the new config API. 
```

Comment:
This seems backwards.  If someone is using this, the old should be retired and perhaps force-assigned to reflect these values.

Reply:
>  If someone is using this, the old should be retired and perhaps force-assigned to reflect these values.

i might not be following what "force reassigned to reflect these values means" (forced in which direction), but fundamentally if someone has config in a secret, that config should take precedence over any default values implied by the introduction of this new api resorce with empty values, no?

and if they have existing tooling that's ignorant of this new api and continues to update the secret, i'd expect those config values to be respected, until such time as we actually remove support for the "secret-based" config api (which, being that it's an api in my mind, we can't do without some sort of deprecation process)

Reply:
Another point is that the current proposal doesn't provide all the config options available via the `support` secret so we can't simply deprecate it IMO.

Reply:
@deads2k based on my understanding after talking with @tremes there is no clear goal yet, to fully replace these config secret, that's why the secret precedence here makes sense. Until there's a clear desire to move out of the config we can switch the priorities, but for now, given only limited set of functionality is supported there I'm ok with this approach.